detection_rules/remote_validation.py

```diff
@@ -0,0 +1,199 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0; you may not use this file except in compliance with the Elastic License
+# 2.0.
+
+from dataclasses import dataclass
+from datetime import datetime
+from functools import cached_property
+from multiprocessing.pool import ThreadPool
+from typing import Dict, List, Optional
+
+import elasticsearch
+from elasticsearch import Elasticsearch
+from marshmallow import ValidationError
+from requests import HTTPError
+
+from kibana import Kibana
+
+from .misc import ClientError, getdefault, get_elasticsearch_client, get_kibana_client
+from .rule import TOMLRule, TOMLRuleContents
+from .schemas import definitions
+
+
+@dataclass
+class RemoteValidationResult:
+    """Dataclass for remote validation results."""
+    rule_id: definitions.UUIDString
+    rule_name: str
+    contents: dict
+    query_results: Optional[dict]
+    engine_results: Optional[dict]
+
+
+class RemoteConnector:
+    """Base client class for remote validation and testing."""
+
+    MAX_RETRIES = 5
+
+    def __init__(self, parse_config: bool = False, **kwargs):
+        es_args = ['cloud_id', 'ignore_ssl_errors', 'elasticsearch_url', 'es_user', 'es_password', 'timeout']
+        kibana_args = [
+            'cloud_id', 'ignore_ssl_errors', 'kibana_url', 'kibana_user', 'kibana_password', 'space', 'kibana_cookie',
+            'provider_type', 'provider_name'
+        ]
+
+        if parse_config:
```

Comment:
What if we dont want to use the config?

Reply:
then you just call the auth methods below directly and pass your kwargs to explicitly get a client